jina/peapods/runtimes/prefetch/base.py

```diff
@@ -100,23 +89,15 @@ async def prefetch_req(num_req, fetch_to):
             for _ in range(num_req):
                 try:
                     if hasattr(request_iterator, '__anext__'):
-                        next_request = await request_iterator.__anext__()
+                        request = await request_iterator.__anext__()
                     elif hasattr(request_iterator, '__next__'):
-                        next_request = next(request_iterator)
+                        request = next(request_iterator)
                     else:
                         raise TypeError(
                             f'{typename(request_iterator)} does not have `__anext__` or `__next__`'
                         )
 
-                    future = get_or_reuse_loop().create_future()
-                    self._message_buffer[next_request.request_id] = future
-                    asyncio.create_task(
-                        self.iolet.send_message(
-                            Message(None, next_request, 'gateway', **vars(self.args))
-                        )
-                    )
-
-                    fetch_to.append(future)
+                    self.handle_request(request=request, fetch_to=fetch_to)
```

Comment:
lets return the task in the `handle_request` and append it to the list here? Otherwise we have to expose the fetch_to list to the child class

Reply:
done